packages/airnode-protocol/test/utils/SelfServeRrpBeaconServerWhitelister.sol.js

```diff
@@ -0,0 +1,363 @@
+  // Fulfill
+  const now = (await hre.ethers.provider.getBlock(await hre.ethers.provider.getBlockNumber())).timestamp;
+  await hre.ethers.provider.send('evm_setNextBlockTimestamp', [now + 1]);
+  const encodedData = 123;
+  encodedTimestamp = now;
+  const data = hre.ethers.utils.defaultAbiCoder.encode(['int256', 'uint256'], [encodedData, encodedTimestamp]);
+  const signature = await airnodeWallet.signMessage(
+    hre.ethers.utils.arrayify(
+      hre.ethers.utils.keccak256(hre.ethers.utils.solidityPack(['bytes32', 'bytes'], [requestId, data]))
+    )
+  );
+  await airnodeRrp
+    .connect(sponsorWallet)
+    .fulfill(
+      requestId,
+      airnodeAddress,
+      rrpBeaconServer.address,
+      rrpBeaconServer.interface.getSighash('fulfill'),
+      data,
+      signature,
+      { gasLimit: 500000 }
+    );
+
+  // Setup another beaconId to read from
+
+  // Compute the expected request ID
+  const requestId2 = hre.ethers.utils.keccak256(
+    hre.ethers.utils.solidityPack(
+      ['uint256', 'address', 'address', 'uint256', 'bytes32', 'address', 'address', 'address', 'bytes4', 'bytes'],
+      [
+        (await hre.ethers.provider.getNetwork()).chainId,
+        airnodeRrp.address,
+        rrpBeaconServer.address,
+        await airnodeRrp.requesterToRequestCountPlusOne(rrpBeaconServer.address),
+        templateId2,
+        roles.sponsor.address,
+        sponsorWalletAddress,
+        rrpBeaconServer.address,
+        rrpBeaconServer.interface.getSighash('fulfill'),
+        beaconParameters,
+      ]
+    )
+  );
+
+  await rrpBeaconServer
+    .connect(roles.updateRequester)
+    .requestBeaconUpdate(templateId2, roles.sponsor.address, sponsorWalletAddress, beaconParameters);
+
+  const now2 = (await hre.ethers.provider.getBlock(await hre.ethers.provider.getBlockNumber())).timestamp;
+  beaconIdExpirationTimeStamp = now2 + 10;
+  await hre.ethers.provider.send('evm_setNextBlockTimestamp', [now2 + 1]);
+  const encodedData2 = 456;
+  encodedTimestamp2 = now2;
+  const data2 = hre.ethers.utils.defaultAbiCoder.encode(['int256', 'uint256'], [encodedData2, encodedTimestamp2]);
+  const signature2 = await airnodeWallet.signMessage(
+    hre.ethers.utils.arrayify(
+      hre.ethers.utils.keccak256(hre.ethers.utils.solidityPack(['bytes32', 'bytes'], [requestId2, data2]))
+    )
+  );
+  await airnodeRrp
+    .connect(sponsorWallet)
+    .fulfill(
+      requestId2,
+      airnodeAddress,
+      rrpBeaconServer.address,
+      rrpBeaconServer.interface.getSighash('fulfill'),
+      data2,
+      signature2,
+      { gasLimit: 500000 }
+    );
+
+  selfServeRrpBeaconServerWhitelister = await selfServeRrpBeaconServerWhitelisterFactory.deploy(
+    rrpBeaconServer.address,
+    [beaconId],
+    [beaconIdExpirationTimeStamp],
+    [beaconId2],
+    [true]
+  );
+
+  const managerRootRole = await accessControlRegistry.deriveRootRole(roles.manager.address);
```

Comment:
Figuring out this step would take me a lot of time. It was good, that you took over this for me.